### PR TITLE
Use official aws cli

### PR DESCRIPTION
**Pull request by Palleas**
Since we're moving cocoadocs to buddybuild's servers, I changed a couple of things: 

* I moved to ruby 2.3.1 as it's already baked into the VMs we use to generate a documentation
* I replaced s3cmd with Amazon's official cli (which we use extensively on buddybuild). The reason is s3cmd doesn't support federation tokens with env vars (cf s3tools/s3cmd#557) and we use that to communicate with our S3 buckets. 

I also have another remaining commit on the side that cleans up the `Rakefile` as we won't be deploying the website the same way.

Please let me know what you think :)

### CODE REVIEW

**Comment by segiddins** [comment]
Could you please update the readme with instructions for installing the different s3 command?

**Comment by Palleas** [comment]
@segiddins done!

File: Rakefile [hunk 1]
```diff
@@ -92,9 +92,12 @@ begin
     end
 
     check_and_install "cloc"
-    check_and_install "s3cmd"
     check_and_install "appledoc"
     check_and_install "carthage"
+
+    # Install pip and AWS
+    `which pip || (curl https://bootstrap.pypa.io/get-pip.py | python)`
+    `sudo pip install awscli`
```
Comment (segiddins) [inline Rakefile]: It appears to be on homebrew as well?

Comment (Palleas) [inline Rakefile]: Oops, fixed. I searched for `aws-cli` and couldn't find anything.  ¯\\_(ツ)_/¯

**Comment by orta** [comment]
Yep - this looks 👍

**Comment by orta** [comment]
I can't transfer ownership of an S3 bucket, just looked, so WRT handling the buckets- would you prefer to be added to the existing one, or to re-setup DNS etc against a different bucket ( and run `s3 sync` to copy existing resources )?

**Comment by Palleas** [comment]
I think it'd be easier to go with the different bucket so we can take care of the S3 bill? 🤔

**Comment by orta** [comment]
That works, you definitely want all the data from the current bucket though - there'll be libraries that you literally couldn't re-parse anymore for whatever reasons ( and trying to re-parse every version of all 26k libraries is has taken months before ) 

So we can figure that out when it's time to switch DNS etc

**Pull request merged**